lychee-lib/src/utils/fragment_checker.rs

```diff
@@ -67,7 +68,17 @@ impl FragmentChecker {
         if fragment.is_empty() || fragment.eq_ignore_ascii_case("top") {
             return Ok(true);
         }
-        let mut fragment_decoded = percent_decode_str(fragment).decode_utf8()?;
+        let mut fragment_candidates = vec![Cow::Borrowed(fragment)];
+        // For GitHub links, add "user-content-" prefix to the fragments.
+        // The following cases cannot be handled unless we simulate with a headless browser:
+        // - markdown files from any specific path (includes "blob/master/README.md")
+        // - "issuecomment" fragments from the GitHub issue pages
+        if url
+            .host_str()
+            .is_some_and(|host| host.ends_with("github.com"))
+        {
+            fragment_candidates.push(Cow::Owned(format!("user-content-{fragment}")));
```

Comment:
You can simplify this to:

```rust
fragment_candidates.push(format!("user-content-{fragment}").into());
```

This works because `String` implements `Into<Cow<'_, str>>`, so `.into()` will automatically convert it to `Cow::Owned(string)`.

Alternatively, if you want to be more explicit about the type:

```rust
fragment_candidates.push(Cow::from(format!("user-content-{fragment}")));
```

Both are cleaner than manually wrapping with `Cow::Owned()`.

See https://doc.rust-lang.org/std/borrow/enum.Cow.html#impl-From%3CCow%3C'a,+str%3E%3E-for-String